plugins/module_utils/image_archive.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 Red Hat | Ansible
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+
+import json
+import tarfile
+
+from ansible.module_utils.common.text.converters import to_native
+
+
+class ImageArchiveInvalidException(Exception):
+    def __init__(self, message, cause):
+        super(ImageArchiveInvalidException, self).__init__(message)
+
+        # Python 2 doesn't support causes
+        self.cause = cause
+
+
+def api_image_id(archive_image_id):
+    """
+    Accepts an image hash in the format stored in manifest.json, and returns an equivalent identifier
+    that represents the same image hash, but in the format presented by the Docker Engine API.
+    """
+
+    return 'sha256:%s' % archive_image_id
+
+
+def archived_image_action(failure_logger, archive_path, current_image_name, current_image_id):
```

Comment:
I don't think this function belongs in this file. It should be part of the docker_image module.

Reply:
Heh I thought about that and waffled on it.

You prefer a standalone function or a method?

Reply:
Considering that most functionality in that module is a method, I would make it a method there.

Reply:
There's some shared functions for stubbing tar files that would be needed by both tests in `modules/` and `module_utils/` and I could create a conftest.py in their parent to share them, but that scopes them across all tests, as opposed to scoping them to just `docker_image`.

Is that OK?

If I leave `def archived_image_action` where it is, the new `conftest.py` isn't needed at all.

Reply:
You can also put unit test code in arbitrary `.py` files and just import it. There's no need to always use `conftest.py` for that.

Reply:
Added a new file `tests/unit/plugins/test_support/docker_image_archive_stubbing.py`